pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,223 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from __future__ import print_function
+
+import numpy as np
+import scipy.constants
+from pyiron.atomistics.master.parallel import AtomisticParallelMaster
+from pyiron_base import JobGenerator
+from sklearn.linear_model import LinearRegression
+from collections import defaultdict
+
+__author__ = "Joerg Neugebauer, Jan Janssen"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Jan Janssen"
+__email__ = "janssen@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+
+eV_div_A3_to_GPa = (
+    1e21 / scipy.constants.physical_constants["joule-electron volt relationship"][0]
+)
+
+def calc_elastic_tensor(strain, stress=None, energy=None, volume=None, rotations=None):
```

Comment:
Nit: This is long and hard to read. It would be helpful to have it refactored into smaller chunks so that `calc_elastic_tensor` read more like

```python
if len(strain)==0:
         raise ValueError('Not enough points')

rotations = _build_all_rotations(rotations)
strain = _expand_strain_list_using_rotations(strain)
if stress is not None and len(stress)*len(rotations)==len(strain):
    coeff = _fit_coeffs_with_stresses(stress, strain, rotations)
elif energy is not None and volume is not None and len(energy)*len(rotations)==len(strain) and len(energy)==len(volume):  
    coeff = _fit_coeffs_with_energies(energy, volume, strain, rotations)
else:
         raise ValueError('Problem with fitting data')
return coeff
```

These functions might need to be further broken down. E.g. in the block I pulled out as `_fit_coeffs_with_energies`, the four-consecutive lines starting with `C=` or the four with `coeff=` are also yellow flags for pulling out a well-named function describing all the math.

Reply:
ok that's true let me see